### DISCUSSION TITLE
Does Embedding Infinispan support rolling upgrades?

### DISCUSSION THREAD

**Discussion by koshox** [infinispan/infinispan#9885] [Q&A]
From [Upgrading Infinispan 13.0](https://infinispan.org/docs/stable/titles/upgrading/upgrading.html#rolling-upgrades) ,I know that the server mode supports rolling upgrades.
Now I want to know if the embedded mode supports rolling upgrades, such as starting some nodes that use a higher version of infinispan to join the cluster？Hope someone can help.

**Comment by tristantarrant** [comment]
No, this is not possible yet. We do have plans for in-place upgrading of
nodes, but it requires some changes to our remote commands

**Reply by koshox** [reply]
Thanks, I still have two questions.
1.Whether the embedded nodes can exist at the same time when the version difference is not large.
2.How to judge，can or can't roll upgrade，For example, how to accurately determine how large the difference between the two versions is, it cannot be exist at the same.

**Reply by ethan-gallant** [reply]
Hey @tristantarrant and thank you for taking the time to reply.

I am also wondering about the compatibility between versions as we are using Infinispan as a sort of database for tracking live location data.  

* Is it planned to have a live upgrade path from v13-v14?
* Is there any solution that we could hack for the time being to support an upgrade in the future? I envision some form of shading hackery to have two versions of the infinispan dependency and do dual groups with a cache listener to sync the new cache.

Thank you for your help! :)

**Reply by Kiiv** [reply]
Hi, is it true also for minor versions or only for major ones ?

**Reply by ryanemerson** [reply]
Infinispan currently provides no guarantees of rolling upgrades between versions major, minor or patch. Patch releases are the most likely to be compatible, however such upgrades can still break if changes are required to Infinispan command internals or JGroup protocols between patch versions.
